internal/pkg/runtime/engine/apptainer/container_linux.go

```diff
@@ -1786,6 +1786,16 @@ func (c *container) addBindsMount(system *mount.System) error {
 			}
 		}
 		if !skipAllBinds && !slice.ContainsString(skipBinds, localtimePath) {
+			if c.engine.EngineConfig.GetSessionLayer() == apptainer.OverlayLayer {
+				content, err := os.ReadFile(localtimePath)
+				if err != nil {
+					return fmt.Errorf("while reading host %s, err: %s", localtimePath, err)
+				}
+				if err := c.session.AddFile(filepath.Join(c.session.Layer.Dir(), localtimePath), content); err != nil {
+					return err
+				}
+			}
```

Comment:
I don't like that all this code is duplicated below.  Can it be put into a function instead?

Reply:
thanks! updated in the latest commit.